src/main/java/com/google/sps/servlets/AddSaleCardServlet.java

```diff
@@ -0,0 +1,173 @@
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.GeoPt;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.sps.data.HttpServletUtils;
+import java.io.IOException;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * When the user submits the form, Blobstore processes the file upload and then forwards the request
+ * to this servlet.
+ */
+@WebServlet("/add-saleCard")
+public class AddSaleCardServlet extends HttpServlet {
+
+  private final static float MAX_DISTANCE = 3000f; // 3 kilometers
+  private final static float MIN_DISTANCE = 0f;
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // TODO: Get if vendor is logged in with auth API
+    boolean vendorStatus = false;
+    
+    if (vendorStatus) {
+      // -- The way we get these values might change. This is just an idea ---------------
+      // Get parameters
+      String businessName = HttpServletUtils.getParameter(request, "businessName", "");
+      boolean hasDelivery = Boolean.parseBoolean(HttpServletUtils.getParameter(request, "hasDelivery", "false"));
+      String startTime = HttpServletUtils.getParameter(request, "startTime", "");
+      String endTime = HttpServletUtils.getParameter(request, "endTime", "");
+      String description = HttpServletUtils.getParameter(request, "description", "");
+      String geoHash = HttpServletUtils.getParameter(request, "geoHash", "");
+      float radius = 0f;
+      float latitude = 0f;
+      float longitude = 0f;
+      GeoPt vendorLocation = new GeoPt(0f, 0f);
+      
+      // Check values are valid format
+      try {
+        LocalTime start = LocalTime.parse(startTime);
+        LocalTime end = LocalTime.parse(endTime);
+        radius = Float.parseFloat(HttpServletUtils.getParameter(request, "radius", "1000"));
+        latitude = Float.parseFloat(HttpServletUtils.getParameter(request, "lat", "360"));
+        longitude = Float.parseFloat(HttpServletUtils.getParameter(request, "long", "360"));
```

Comment:
What the third parameter of `getParameter()` is ? I did not understand why do we need to pass 1000,360,360. Maybe add a comment for that will be useful.

Reply:
The third parameter is a default value, in case the request's parameter does not exist or is empty.
In this case, for radius the default value is 1000 (1 km) decided arbitrarily.
But for latitude and longitude is 360 to avoid having an extra if-statement to check if they are inside the range of a valid latitude and longitude. Since 360 is outside this range, when trying to create a GeoPt with these values, the GeoPt constructor will throw an `IllegalArgumentException`, which is caught.